MetaMask/metamask-extension
# Don't allow Snaps import duplicate accounts

This function will be used by the `SnapKeyring` to check if an address is already present in the `KeyringController` and prevent duplicate accounts from being added.

## **Description**

This function will be used by the `SnapKeyring` to check if an address is already present in the `KeyringController` and prevent duplicate accounts from being added.

## **Manual testing steps**

1. Install Flask from this PR and import your SRP.
2. Go the: "Accounts list" > "Account 1" > "Account details".
3. Click on "Show private key" and copy the private key.
4. Go to the [SSK dapp][ssk-dapp] and connect the dapp and install the SSK Snap.
5. Expand "Import account" and paste the private key.
6. Click on "Import Account".

    This should return the following error message:

    ```json
    {
      "code": -32603,
      "message": "Account address '<ACCOUNT 1 ADDRESS>' already exists",
      "data": {
        "originalError": {}
      }
    }
    ```

7. Go to your "Accounts list", the "Account 1" shouldn't have the "Snap" label.
8. Go back to the SSK dapp and reload it.
9. Due to the SSK ignoring the above error, it will display the account as imported.
10. Remove the account from the SSK.
11. The account should disappear from the SSK but still be present in the "Accounts list"

## **Screenshots/Recordings**

https://github.com/MetaMask/metamask-extension/assets/68558152/ed06cc56-77d0-42d2-87d6-42caab78f6e9

## **Related issues**

Fixes: https://github.com/MetaMask/accounts-planning/issues/61

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**


[ssk-dapp]: https://metamask.github.io/snap-simple-keyring/latest/
